### ISSUE TITLE
WIFI does not connect

### ISSUE THREAD

**Issue by pixeldoc2000** [patience4711/read-APSystems-YC600-QS1-DS3#101]
Can't get WIFI to connect anymore.

My ECU was running fine for about a year+ on my wemos D1 mini clone. I belive I started with v9.3 .

Latest Update from 9.6 to 9.6c failed. I've had to reflash my ESP8266. After reflash I can't get it to connect to my Wifi / WLAN anymore, no matter what.

What I have tried already:
- tried other ESP8266 modules -> same Problem
   - Same two ESP8266 connects to my WIFI (without changed Settings) without any issue with ESPHOME Firmware
   - Nodemcu
   - wemos d1 mini
- Change WIFI Settings
   - Disabled 5G
   - Changed to WPA from WPA2/3
   - Changed Password und SSID
- Tried several Firmware Version from v9.8 to v9.3
- Erased Flash several times via different tools
- Write Empty Flash
- Verify Flash and Flash Size

Looking info compiling firmware myself with more serial debug messages enabled to get a clue why it is not connecting.

Sadly the (wifi) source code is rather hard to follow (spagetti code).

Any suggestions what else to try?

**Comment by kentbrock** [comment]
Hi,

have you tried a clean up with the Espressif Tool?

https://www.espressif.com/en/support/download/other-tools

Sometimes I face this issue in other projects. A factory wipe always helps.

BR

**Comment by pixeldoc2000** [comment]
@kentbrock 

Do you mean "Flash Download Tools" by Espressif? I don't see a way to do a factory reset.

But I already did a couple of erase_flash with esptool: `esptool --chip esp8266 erase_flash` which should do (almost) the same?

**Comment by joffito** [comment]
try a power cycle and after powering on press the flash button. hold it till the led lights on and off (might be several times). the  release flash button and power cycle.

**Comment by patience4711** [comment]
I am going to close this, it is a duplication of https://github.com/patience4711/read-APSystems-YC600-QS1-DS3/issues/66

**Comment by pixeldoc2000** [comment]
@patience4711 I have seen that one, but the Solution to erase the flash did not work for me. That's why I opened a new issue.

@joffito did that the power cycle and try to enter flash mode connecting D3 / GIO0 and GND (d1 mini has no flash button) but the led light did not come on. I had to reflash the wemos d1 again. Did not make a differents.

Serial Log:
```
[12:20:26]/wifiCon requested
[12:20:26]send confirm page  
[12:20:32]could not connect, try again
[12:23:33]lijst = 
[12:23:33] 
[12:23:33]  <div><a href="#p" onclick="c(this)">FOO</a>&nbsp;<span class="q ">-61dBm</span></div>
[12:23:33]   
[12:23:33]  <div><a href="#p" onclick="c(this)">BAA</a>&nbsp;<span class="q ">-86dBm</span></div>
[12:23:33]  
[12:23:43]/wifiCon requested
[12:23:43]send confirm page  
[12:23:50]could not connect, try again
```
This will continue infinitely.

Look like the WIFI credentials are not saved. After reset, it will always enter "Portal Loop".

**Comment by joffito** [comment]
just get yourself a 5 $ NodeMCU 8266 and enjoy ;-)

**Comment by pixeldoc2000** [comment]
@joffito I tested a nodemcu, same issue, so no joy for me.

Simple Test Webserver with Arduino Example works also on my wemos d1 mini and nodemcu (like ESPHOME), so no Hardware Error.

So it must be a problem with the ECU Firmware.

**Comment by swbouman** [comment]
It’s too bad that you experience these errors. Did you first erase whole esp? Sometimes this gives an error. 

What version did you flash? Did you try more versions?

**Comment by pixeldoc2000** [comment]
@swbouman as I wrote before, yes erased the flash multiple time on multiple Devices, tried multiple ESP (they all work with every other Firmware but ECU), tried multiple Versions of ECU Firmware.
